components/service/glean/src/test/java/mozilla/components/service/glean/GleanTest.kt

```diff
@@ -288,4 +289,32 @@ class GleanTest {
         gleanSpy.initialized = false
         gleanSpy.handleEvent(Glean.PingEvent.Default)
     }
+
+    @Test
+    fun `Don't send metrics if metrics disabled`() {
+        val server = MockWebServer()
+        server.enqueue(MockResponse().setBody("OK"))
+
+        EventsStorageEngine.clearAllStores()
+
+        val realClient = Glean.httpPingUploader
+        val testConfig = Glean.configuration.copy(
+            serverEndpoint = "http://" + server.hostName + ":" + server.port
+        )
+        Glean.httpPingUploader = HttpPingUploader(testConfig)
+        Glean.setMetricsEnabled(false)
+
+        try {
+            Glean.handleEvent(Glean.PingEvent.Background)
+
+            val requests: MutableMap<String, String> = mutableMapOf()
+
+            val request = server.takeRequest(2, TimeUnit.SECONDS)
```

Comment:
Please leave a comment mentioning that this test will only work since we're faking the dispatchers with the `@Rule` above, otherwise it would fail due to async weirdness!